src/storage/btree.c

```diff
@@ -22397,23 +22397,58 @@ btree_rv_redo_global_unique_stats_commit (THREAD_ENTRY * thread_p, const LOG_RCV
   return NO_ERROR;
 }
 
+/*
+ * btree_rv_data_unpack_btid_and_stats - unpack btree identifier and statistics
+ *
+ * NOTE: keep pack/upack order consistent (same as unique stats struct order)
+ */
 void
-btree_rv_data_get_btid_and_stats (const LOG_RCV & rcv, BTID & btid, log_unique_stats & stats)
+btree_rv_data_unpack_btid_and_stats (const LOG_RCV & rcv, BTID & btid, log_unique_stats & stats)
 {
   assert (rcv.length >= (3 * OR_INT_SIZE) + OR_BTID_ALIGNED_SIZE);
 
   const char *datap = rcv.data;
   OR_GET_BTID (datap, &btid);
   datap += OR_BTID_ALIGNED_SIZE;
-  stats.num_keys = OR_GET_INT (datap);
+  stats.num_nulls = OR_GET_INT (datap);
   datap += OR_INT_SIZE;
   stats.num_oids = OR_GET_INT (datap);
   datap += OR_INT_SIZE;
-  stats.num_nulls = OR_GET_INT (datap);
+  stats.num_keys = OR_GET_INT (datap);
   datap += OR_INT_SIZE;
   assert ((datap - rcv.data) == (size_t) rcv.length);
 }
 
+/*
+ * btree_rv_data_pack_btid_and_stats - pack btree identifier and statistics
+ *
+ * NOTE: keep pack/upack order consistent (same as unique stats struct order)
+ */
+void
+btree_rv_data_pack_btid_and_stats (const BTID * btid, int nulls, int oids, int keys, RECDES & record_descriptor)
```

Comment:
A record descriptor is not required; only packing & the packed length. I think the existing code was misusing the recdes.
```suggestion
btree_rv_data_pack_btid_and_stats (const BTID * btid, int nulls, int oids, int keys, char * data, int &length)
```
Also size_t type is an option for length.